test/simulator/timestamp/src/session_simulator.cpp

```diff
@@ -85,13 +90,33 @@ session_simulator::rollback_transaction()
     _txn_running = false;
 }
 
-void
-session_simulator::commit_transaction()
+int
+session_simulator::commit_transaction(const std::string &config)
 {
     /* Make sure that the transaction from this session is running. */
     assert(_txn_running);
 
+    timestamp_manager *ts_manager = &timestamp_manager::get_timestamp_manager();
+    std::map<std::string, std::string> config_map;
+
+    ts_manager->parse_config(config, config_map);
+
+    auto pos = config_map.find("commit_timestamp");
+    if (pos != config_map.end()) {
+        for (auto &ch : pos->second)
+            if (!std::isxdigit(ch))
+                WT_SIM_RET_MSG(EINVAL, "Illegal commit timestamp: invalid hex value.");
```

Comment:
Should we have this check in session_simulator::decode_timestamp_config_map as well?

Reply:
Yeah will do.